### ISSUE TITLE
Claiming my work as your own.

### ISSUE THREAD

**Issue by obmarg** [smeevil/flycheck-elixir-dogma#1]
I noticed that you forked my original flycheck-elixir-dogma repository last week:

![image](https://cloud.githubusercontent.com/assets/556490/13393484/a84c0c3c-ded9-11e5-846b-90a0e2c5db46.png)

I decided to have a look today and see what changes you'd made, when I noticed that you'd deleted that fork and re-uploaded the repository with your name instead of mine.  I don't really mind, but it'd be courteous to at least mention my name somewhere.

**Comment by smeevil** [comment]
Yeah of course, I’m busy in the process to learn Emacs lisps and was just messing around with the flychecker templates and all.
I still have a lot of problems figuring out why it bails on the require of flycheck.

Since I’m also busy with integrating the other tools like Mix Test and Credo into I made all the files the same.
It’s not my intention at all to rip off your template without contribution! 
As soon as I’m done with figuring out how to fix the errors and get the templates to parse successful I was planning to refork, update and create a pull request to your original :)

Thanks for contacting me though !
Gerard.

**Comment by obmarg** [comment]
Ah, fair enough - that's alright then :)

It's been a while since I actually used this, and having just tried again it seems I'm also getting a problem around the require on flycheck.  I'll see if I can figure it out as well.

**Comment by obmarg** [comment]
Ok, so I've fixed that problem on my local machine.  Are you using spacemacs or plain emacs?

**Comment by smeevil** [comment]
Spacemacs

**Comment by obmarg** [comment]
I had to make some changes to my spacemacs layer here: https://github.com/obmarg/dotfiles/commit/2d2e151006c8e8a0c26ee70281965afac9226e94 to get it working with the latest spacemacs.  You might want to try the same.

**Comment by smeevil** [comment]
also, if you are interested, here are the two other flychecks : 


https://github.com/smeevil/flycheck-elixir-credo

And one that I’m working on to get the results of mix test into flycheck :

https://github.com/smeevil/flycheck-elixir-testresult

This one has a helper file and a formatter with it for elixir.

**Comment by obmarg** [comment]
Nice - I'll definitely check them out when I get a minute.  

I have been meaning to put together a flycheck for compiling elixir, though not sure when/if I'll get round to it - one of the problems with this dogma one is it falls badly when you have an elixir syntax error.

**Comment by smeevil** [comment]
yes, I’ve been trying to look into that as well. 
The other issue I notice is when you use the source-inplace it creates a flycheck_<filename> file in the directory that it uses to run the check on. Ot seems it creates/renews this file on every change (keypress) in the file. So when you have a mix test.watch running, it will go berserk on it because of the auto reload kicking in all the time. As well as Credo complaining about duplicate code since that also parses the new temp file. 

So in the end it would be a lot better if we could just pipe the stdin to dogma / credo.
Maybe also run it as a sort of deamon to speed things up…

Lots of idea’s anyways :)

**Comment by obmarg** [comment]
Ah yes - I don't have any other things watching my directories, but I can see how that would be a problem.  It's relatively trivial to get flycheck to pipe via stdin, but that wasn't an option in dogma when I last checked.

**Comment by smeevil** [comment]
I’m working on getting dogma to accept stdin, current progress :



Its very rough at the moment, but at least it proves that its working.

**Comment by smeevil** [comment]
Got it done, and now waiting for the merge :)
https://github.com/smeevil/dogma/commit/f965b00c4bb48427467bf04d65f8fb9754531d03 <https://github.com/smeevil/dogma/commit/f965b00c4bb48427467bf04d65f8fb9754531d03>

**Comment by obmarg** [comment]
Nice one, that will be awesome once it's merged. Was doing some elixir last
night and those temporary files really are a pain.
